experiments/AMIP/modular/components/land/bucket_init.jl

```diff
@@ -49,77 +51,87 @@ multiple dispatch on `surface_fluxes`.
 struct CoupledAtmosphere{FT} <: AbstractAtmosphericDrivers{FT} end
 
 """
-    surface_fluxes(Y, p, 
-                    t::FT,
-                    parameters::P,
-                    atmos::PA,
-                    ) where {FT <: AbstractFloat, P <: BucketModelParameters{FT},  PA <: CoupledAtmosphere{FT}}
+    surface_fluxes(atmos::CoupledAtmosphere{FT},
+                    model::BucketModel{FT},
+                    Y,
+                    p,
+                    ) where {FT <: AbstractFloat}
 
 Computes the turbulent surface fluxes terms at the ground for a coupled simulation.
 """
 function ClimaLSM.Bucket.surface_fluxes(
-    Y::ClimaCore.Fields.FieldVector,
-    p::ClimaCore.Fields.FieldVector,
-    t::FT,
-    parameters::BucketModelParameters{FT, P},
     atmos::CoupledAtmosphere{FT},
-) where {FT <: AbstractFloat, P}
-    # coupler has done its thing behind the scenes already
+    model::BucketModel{FT},
+    Y,
+    p,
+    _...,
+) where {FT <: AbstractFloat}
     return (turbulent_energy_flux = p.bucket.turbulent_energy_flux, evaporation = p.bucket.evaporation)
 end
 
 
 
 """
-    net_radiation(Y, p, 
-                    t::FT,
-                    parameters::P,
-                    radiation::PR, p,
-                    ) where {FT <: AbstractFloat, P <: BucketModelParameters{FT}, PR <: CoupledRadiativeFluxes{FT}}
+    net_radiation(radiation::CoupledRadiativeFluxes{FT},
+                    model::BucketModel{FT},
+                    Y,
+                    p,
+                    _...,
+                    ) where {FT <: AbstractFloat}
 
 Computes the net radiative flux at the ground for a coupled simulation.
 """
 function ClimaLSM.Bucket.net_radiation(
+    radiation::CoupledRadiativeFluxes{FT},
+    model::BucketModel{FT},
     Y::ClimaCore.Fields.FieldVector,
     p::ClimaCore.Fields.FieldVector,
-    t::FT,
-    parameters::BucketModelParameters{FT, P},
-    radiation::CoupledRadiativeFluxes{FT},
-) where {FT <: AbstractFloat, P}
+    _...,
+) where {FT <: AbstractFloat}
     # coupler has done its thing behind the scenes already
     return p.bucket.R_n
 end
 
 
 """
-    ClimaLSM.Bucket.surface_air_density(p, atmos::CoupledAtmosphere)
-
-an extension of the bucket model method which returns the surface air 
+    ClimaLSM.surface_air_density(
+                    atmos::CoupledAtmosphere{FT},
+                    model::BucketModel{FT},
+                    Y,
+                    p,
+                    _...,
+                )
+an extension of the bucket model method which returns the surface air
 density in the case of a coupled simulation.
 """
-function ClimaLSM.Bucket.surface_air_density(p, atmos::CoupledAtmosphere)
-    # coupler has filled this in
+function surface_air_density(
+    atmos::CoupledAtmosphere{FT},
+    model::BucketModel{FT},
+    Y,
+    p,
+    _...,
+) where {FT <: AbstractFloat}
     return p.bucket.ρ_sfc
```

Comment:
The coupler has still set this value, correct? On our side, we compute it ourselves (PrescribedAtmos). But for anything dispatching off of CoupledAtmos, the coupler has to set that field

Reply:
We have a function `set_ρ_sfc!` [here](https://github.com/CliMA/ClimaCoupler.jl/blob/main/experiments/AMIP/modular/components/flux_calculator.jl#L12), but I don't see it actually get called. @LenkaNovak can you confirm if we set `ρ_sfc`?

Reply:
Yes, at the moment we're using Atmos's `set_surface_thermo_state`, which calculates `ρ_sfc` internally, so in fact this variable doesn't need to exist in the coupler anymore. I was going to remove it as part of the generalized interface SDI but I won't object to deleting it here.

Reply:
Discussed offline - we're going to remove `ρ_sfc` in the generalized interface SDI to avoid scope creep here.

Reply:
sounds good, thank you for checking!